### PR TITLE
Add term relationship type model

### PR DESCRIPTION
**Pull request by atpw25**

### CODE REVIEW

File: src/mdm-term-relationship-type.model.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  label?: string;
+}
+
+export type TermRelationshipTypeDetail =
+  TermRelationshipType
+  & Navigatable
+  & Securable
+  & Historical;
+
+export type TermRelationshipTypeIndexResponse = MdmIndexResponse<TermRelationshipType & Navigatable>;
```
Comment (pjmonks) [inline src/mdm-term-relationship-type.model.ts:39]: If `breadcrumbs` is returned from both the index and `GET` endpoints, shouldn't `Navigatable` just be an extension of `TermRelationType`?

```ts
export interface TermRelationshipType extends Navigatable {
}
```

File: src/mdm-term-relationship-type.model.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+import { Historical, Securable } from './mdm-model-types.model';
+import { MdmIndexResponse, MdmResponse, Navigatable, Uuid } from './mdm-common.model';
+
+export type TermRelationshipTypeType = 'TermRelationshipType';
+
+export interface TermRelationshipType {
+  [key: string]: any;
+  id?: Uuid;
+  domainType: TermRelationshipTypeType;
```
Comment (pjmonks) [inline src/mdm-term-relationship-type.model.ts:27]: Two things here:

1. `domainType` feels like it should be of type `CatalogueItemDomainType`, meaning `TermRelationshipType` should be added to that enum
2. Then you can extend `TermRelationshipType` with the `CatalogueItem` interface.

**Review by pjmonks — CHANGES_REQUESTED** [review]
Could you look over the comments I made please? Also, this PR is missing the build output - currently we have to commit the output from `npm run build` as well.

**Comment by atpw25** [comment]
Updated. Also included build output.

**Review by pjmonks — APPROVED** [review]
All changes are good, thanks. I will merge once Jenkins allows it.

**Pull request merged**